### PR TITLE
refactor(validation): finalize packet 14 fail-closed cutover

### PR DESCRIPTION
**Pull request by jamesryancooper**
Policy reference: `.octon/framework/agency/practices/pull-request-standards.md`
Reminder: Run `@codex review`.

## What

Implements the Packet 14 big-bang cutover for validation, fail-closed
publication, quarantine handling, and staleness enforcement across the
runtime-facing effective families.

## Why

Packet 14 was the last missing operational contract tying the super-root
publication surfaces together. This PR makes the fail-closed model,
publication receipts, reduced locality quarantine behavior, and the archived
proposal closeout authoritative in code and docs.

No-Issue: internal Packet 14 harness cutover and proposal closeout.

## How

Promotes Packet 14 as one clear-break change set by adding the new
`validate-runtime-effective-state.sh` umbrella gate, introducing retained
publication receipts under `state/evidence/validation/publication/**`,
bumping the extension/locality/capability publication schemas, and updating
the locality/extension/capability publishers plus validators to enforce the
new trust model.

The cutover also archives the Packet 14 proposal package, records ADR 058 and
migration evidence, and refreshes generated effective outputs plus host
projections so the durable repo state matches the promoted contract.

## Profile Selection Receipt

- Semantic version source(s): `/.octon/octon.yml`
- Release state (`pre-1.0` or `stable`): `pre-1.0`
- `change_profile` (`atomic` or `transitional`): `atomic`
- Hard-gate facts (downtime, coordination, migration/backfill, rollback, blast radius, compliance): internal harness control-plane change; no staged coexistence or backfill required; rollback is full revert; blast radius spans publication scripts, validators, generated outputs, docs, and proposal lifecycle state; must fail closed on stale effective outputs, raw-input leakage, invalid root/class-root state, native/extension collisions, and quarantined repo snapshots
- Tie-break status: `atomic` selected without exception
- `transitional_exception_note` (required when `pre-1.0` + `transitional`): `n/a`

## Implementation Plan

- Workstreams and scope:
  - add the Packet 14 runtime-effective trust gate
  - add retained publication receipts and schema/docs for them
  - update extension/locality/capability publication and validation semantics
  - refresh generated effective outputs and scaffolds
  - archive the Packet 14 proposal with ADR and migration evidence
- Public interfaces/types/contracts affected:
  - `validate-runtime-effective-state.sh`
  - `octon-validation-publication-receipt-v1`
  - extension control/publication schemas: `active/quarantine v3`, `catalog/artifact-map/generation-lock v4`
  - locality publication schemas: `quarantine/effective/artifact-map/generation-lock v2`
  - capability routing publication schemas: `effective/artifact-map/generation-lock v3`
- Test scenarios:
  - reduced locality publication on scope-local quarantine
  - extension receipt emission, stale generation mismatch, and native/extension collision failure
  - capability publication degraded upstream generations and quarantined-scope routing exclusion
  - umbrella runtime-effective gate failures for raw-input leakage, stale outputs, missing locks, cross-family mismatch, and bad class-root bindings
  - clean-only `repo_snapshot` behavior when extension quarantine is present
- Assumptions/defaults:
  - full Packet 14 closeout lands in one atomic PR
  - `repo_snapshot` remains fail-closed on any extension quarantine
  - publication receipts remain retained evidence under `state/**`
  - no compatibility shim or raw-input fallback is introduced

## Impact Map (code, tests, docs, contracts)

- Code:
  - `.octon/framework/assurance/runtime/_ops/scripts/**`
  - `.octon/framework/orchestration/runtime/_ops/scripts/**`
  - `.octon/framework/capabilities/_ops/scripts/publish-capability-routing.sh`
- Tests:
  - `.octon/framework/assurance/runtime/_ops/tests/test-validate-{locality,extension,capability,runtime-effective}-state.sh`
  - `.octon/framework/assurance/runtime/_ops/tests/test-export-harness.sh`
  - `.octon/framework/assurance/runtime/_ops/tests/test-validate-{extension-pack,host-projections}.sh`
- Docs:
  - `.octon/README.md`
  - `.octon/instance/bootstrap/START.md`
  - `.octon/framework/cognition/_meta/architecture/**`
  - `.octon/instance/cognition/decisions/058-validation-fail-closed-quarantine-staleness-atomic-cutover.md`
- Contracts:
  - effective publication schemas under `.octon/framework/cognition/_meta/architecture/generated/effective/**/schemas/`
  - control-state schemas under `.octon/framework/cognition/_meta/architecture/state/control/schemas/`
  - publication receipt schema under `.octon/framework/cognition/_meta/architecture/state/evidence/validation/publication/schemas/`

## Compliance Receipt

- [x] Selected exactly one execution profile before planning/implementation.
- [x] Applied release-maturity gate from semantic version.
- [x] Enforced pre-1.0 default (`atomic`) unless hard gates required `transitional`.
- [x] Included `transitional_exception_note` when required.
- [x] Included tie-break escalation when applicable.
- [x] Updated impacted contracts/docs/tests.
- [x] Honored charter change-control constraint (no direct principles charter edits without override evidence).

## Exceptions/Escalations

- `alignment-check.sh --profile harness` needed an escalated rerun in this
  environment to refresh `.codex/**` host projections cleanly.
- No open exceptions remain.

## Tradeoffs

This keeps all Packet 14 changes atomic, which makes the diff larger and
includes retained publication receipts for the intermediate publish runs that
occurred during verification. The tradeoff is deliberate: the repo keeps the
full evidence trail rather than rewriting retained state to look smaller.

## Testing

- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-extension-pack-contract.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-locality-publication-state.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-extension-publication-state.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-capability-publication-state.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-runtime-effective-state.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-validate-host-projections.sh`
- `bash .octon/framework/assurance/runtime/_ops/tests/test-export-harness.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-proposal-standard.sh --package .octon/inputs/exploratory/proposals/.archive/architecture/validation-fail-closed-quarantine-staleness`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-architecture-proposal.sh --package .octon/inputs/exploratory/proposals/.archive/architecture/validation-fail-closed-quarantine-staleness`
- `bash .octon/framework/assurance/runtime/_ops/scripts/alignment-check.sh --profile harness`

## Rollout

n/a. This is an internal harness control-plane cutover with full-revert
rollback if needed.

## Convivial Purpose Check

- [x] Feature expands genuine user capability (not synthetic engagement).
- [x] Attention and interruption behavior are justified and user-controllable.
- [x] No manipulative patterns or dark-pattern mechanics are introduced.
- [x] Data collection/extraction risk is minimal and explicitly justified.

## Risk Rubric

- Risk class: [ ] Trivial [ ] Low [ ] Medium [x] High
- Rollback plan: revert the two branch commits and republish the pre-cutover
  effective outputs
- Rollback handle: `git revert 67fb21156 58714ae47`
- Flags changed (name, owner, expiry, rollout): none
- Autonomy eligibility: [ ] autonomy:auto-merge [x] autonomy:no-automerge

## Contracts and Threat Model

- OpenAPI/JSON-Schema changes:
  - publication receipt schema added
  - extension/locality/capability generated schema versions bumped
  - extension/locality control-state schema versions bumped
- Threat model update/link:
  - fail-closed trust boundary tightened for stale generations, raw-input
    leakage, bad class-root state, and native/extension collisions

## Observability and Performance

- Traces/logs/metrics for changed flows:
  - retained publication receipts now record each runtime-facing effective
    publication under `state/evidence/validation/publication/**`
- Representative traces for high risk changes:
  - see the Packet 14 migration bundle under
    `.octon/state/evidence/migration/2026-03-20-validation-fail-closed-quarantine-staleness-cutover/`
- Performance or bundle impact:
  - minor validator/publisher overhead from receipt generation and extra scope
    evidence hashing

## License and Provenance

- New dependencies and licenses: none
- Generated code/templates provenance notes:
  - refreshed generated effective outputs and retained publication receipts are
    produced by the in-repo Packet 14 publishers and validators

## Checklist

- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [x] All review conversations resolved

## Screenshots / Notes

No UI changes. One unrelated untracked file remains intentionally out of this
PR: `.octon/inputs/exploratory/proposals/architecture/migration-rollout/resources/octon_packet_15_migration_and_rollout.md`.

### CODE REVIEW

File: .octon/framework/assurance/runtime/_ops/scripts/validate-extension-publication-state.sh [hunk 1]
```diff
@@ -126,6 +152,42 @@ main() {
...
+    {
+      yq -r '.skills[]?.id // ""' "$FRAMEWORK_SKILLS_MANIFEST" 2>/dev/null || true
+      yq -r '.skills[]?.id // ""' "$INSTANCE_SKILLS_MANIFEST" 2>/dev/null || true
+    } | awk 'NF' | LC_ALL=C sort -u
+  )"
+  collision_lines="$(
+    {
+      yq -r '.packs[]? as $pack | $pack.routing_exports.commands[]? | ["command", $pack.pack_id, .capability_id] | @tsv' "$CATALOG_FILE" 2>/dev/null || true
+      yq -r '.packs[]? as $pack | $pack.routing_exports.skills[]? | ["skill", $pack.pack_id, .capability_id] | @tsv' "$CATALOG_FILE" 2>/dev/null || true
+    } | while IFS=$'\t' read -r kind pack_id capability_id; do
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/assurance/runtime/_ops/scripts/validate-extension-publication-state.sh:172]: Addressed in `c676e28b0`.

`publish-extension-state.sh` now enforces native command/skill collisions before writing the published extension state, and the validator remains as a backstop. I also refreshed the generated Packet 14 outputs against `0.5.2` in `9f95139e4`.

File: .octon/framework/orchestration/runtime/_ops/scripts/publish-locality-state.sh [hunk 2]
```diff
@@ -485,65 +722,116 @@ main() {
...
+  done
+
+  manifest_sha="$(hash_file "$LOCALITY_MANIFEST")"
+  registry_sha="$(hash_file "$LOCALITY_REGISTRY")"
+  quarantine_sha="$(hash_file "$quarantine_tmp")"
+  if command -v shasum >/dev/null 2>&1; then
+    quarantine_generation_sha="$(yq -o=json '.records // []' "$quarantine_tmp" 2>/dev/null | tr -d '\n' | shasum -a 256 | awk '{print $1}')"
+  else
+    quarantine_generation_sha="$(yq -o=json '.records // []' "$quarantine_tmp" 2>/dev/null | tr -d '\n' | sha256sum | awk '{print $1}')"
+  fi
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/orchestration/runtime/_ops/scripts/publish-locality-state.sh:760]: Addressed in `58714ae47`.

`publish-locality-state.sh` now computes `quarantine_generation_sha` from the stable record payload rather than the timestamped quarantine file, so noop republishes keep the same `locality-*` generation ID when the quarantined scope set and reason codes do not change.
